util/globals.py

```diff
@@ -48,6 +48,30 @@ def get_macros(self, ioc_name):
                 macros[key] = value
         return macros
 
+    def get_values_of_macro(self, macro_name):
+        """
+        Get the values of all macros of a given name.
+        :param macro_name: The name of the macro to search for
+        :return:  A dictionary of all instances of the matching values and the values.
+        """
+        lines = self.get_lines()
+        macros = dict()
+        for line in lines:
+            if line.__contains__(macro_name):
+                line = line.split("__")[1]
+                key, value = line.split("=")
+                macros[key] = value
+        return macros
+
+    def is_in_sim_mode(self):
+        """
+        :return: TRUE if any simulation flags are set
+        """
+        has_recsim = "1" in self.get_values_of_macro("RECSIM").values()
+        has_devsim = "1" in self.get_values_of_macro("DEVSIM").values()
+        has_simulate = "1" in self.get_values_of_macro("SIMULATE").values()
+        return has_recsim or has_devsim or has_simulate
```

Comment:
Currently this test isn't working if I put `MYIOC__RECSIM=1` in DEMO's `globals.txt`

The reason is because there's a trailing `\n` on the end of the line, so the macro value is being captured as `1\n` and therefore not getting caught properly. I think the best fix here is to modify the `get_lines` method to call `.strip()` on each line before returning it - this will get rid of any leading or trailing whitespace.